nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ListenTCPRecord.java

```diff
@@ -284,95 +286,57 @@ public void onScheduled(final ProcessContext context) throws IOException {
-        final SocketChannelRecordReader socketRecordReader = pollForSocketRecordReader();
-        if (socketRecordReader == null) {
-            return;
-        }
-
-        if (socketRecordReader.isClosed()) {
-            getLogger().warn("Unable to read records from {}, socket already closed", new Object[] {getRemoteAddress(socketRecordReader)});
-            IOUtils.closeQuietly(socketRecordReader); // still need to call close so the overall count is decremented
+        NetworkRecordReader recordReader = pollForRecordReader();
+        if (recordReader == null) {
             return;
         }
 
         final int recordBatchSize = context.getProperty(RECORD_BATCH_SIZE).asInteger();
         final String readerErrorHandling = context.getProperty(READER_ERROR_HANDLING_STRATEGY).getValue();
         final RecordSetWriterFactory recordSetWriterFactory = context.getProperty(RECORD_WRITER).asControllerService(RecordSetWriterFactory.class);
 
-        // synchronize to ensure there are no stale values in the underlying SocketChannel
-        synchronized (socketRecordReader) {
+        // synchronize to ensure there are no stale values in the underlying SocketChannel - is this still necessary?
+        synchronized (recordReader) {
             FlowFile flowFile = session.create();
             try {
-                // lazily creating the record reader here
-                RecordReader recordReader = socketRecordReader.getRecordReader();
-                if (recordReader == null) {
-                    recordReader = socketRecordReader.createRecordReader(getLogger());
-                }
-
                 Record record;
                 try {
-                    record = recordReader.nextRecord();
+                    record = recordReader.getRecordReader().nextRecord(); // TODO: This nextRecord call needs a timeout of some kind
                 } catch (final Exception e) {
-                    boolean timeout = false;
-
-                    // some of the underlying record libraries wrap the real exception in RuntimeException, so check each
-                    // throwable (starting with the current one) to see if its a SocketTimeoutException
-                    Throwable cause = e;
-                    while (cause != null) {
-                        if (cause instanceof SocketTimeoutException) {
-                            timeout = true;
-                            break;
-                        }
-                        cause = cause.getCause();
-                    }
-
-                    if (timeout) {
-                        getLogger().debug("Timeout reading records, will try again later", e);
-                        socketReaders.offer(socketRecordReader);
-                        session.remove(flowFile);
-                        return;
-                    } else {
-                        throw e;
-                    }
+                    throw e;
```

Comment:
I'm not necessarily saying this is wrong, but doesn't this change the behavior a bit?

In the previous logic, if it detected a timeout when trying to read, it would leave the connection open and try again later, now it is throwing the exception which will go to the outer catch block that closes the exception.

Reply:
Concurring with @bbende, it looks like this may need to be revisited.  If SocketTimeoutExceptions are no longer thrown, then this makes sense, but if they are thrown, something similar to this needs to be maintained.

Reply:
At the moment I do not believe a SocketTimeoutException can occur in this flow of code anymore. Any timeout exceptions should be thrown by the Netty server by netty threads, which presumably aren't logged in the ListenTCPRecord bulletin.